src/js/jsx/search/SearchBar.jsx

```diff
@@ -407,20 +420,24 @@ define(function (require, exports, module) {
          * Perform action based on ID
          *
          * @private
-         * @type {Datalist~onInputChange}
+         * @param {string} id of the selected option
          */
         _handleChange: function (id) {
+            var filters = this.state.filter,
+                filtersString,
+                idSplit,
+                type;
+
             if (id === null) {
+                type = "";
                 this.props.dismissDialog();
                 return;
+            } else {
+                idSplit = id.split("-");
```

Comment:
For future reference, I really hate how we have two separate data items encoded in a single concatenated string. This is a terrible programming idiom. This doesn't need to be fixed now, but it should be fixed...